tests/TDD.md

```diff
@@ -0,0 +1,91 @@
+## Test Driven Development (TDD) for ODO
+
+What’s TDD?  In simple terms, WRITE TESTS FIRST
+
+#### TDD Workflow
+
+Under TDD workflow, we write tests before writing feature code
+The tests written will serve as a blueprint for writing feature code
+
+1. Add a test
+1. Run all tests and see if the new test fails
+1. Write the code
+1. Run tests
+1. Rewrite code incorporating any changes, and add new tests
+
+#### Purpose
+
+1. Writing tests before implementing the functionality will enable us to write tests covering more scenarios
+1. A feature with a good set of cases is also the most robust, thereby helping us improve the quality of the product
+1. Would help us address the test coverage backlog we have
```

Comment:
don't really need this part. already covered the purpose in the first sentence of this document